### ISSUE TITLE
variable-naming-style toml configuration: only a single style is allowed?

### ISSUE THREAD

**Issue by ramarivera** [PyCQA/pylint#7352]
### Question

When running pylint from VS code, with the following config:

```
[tool.pylint.BASIC]
variable-naming-style = ["UPPER_CASE", "snake_case"]
```

only the following error is reported

````
[ERROR 2022-7-25 2:41:18.138]: Linter 'pylint' failed to parse the output '. [SyntaxError: Unexpected end of JSON input
	at JSON.parse (<anonymous>)
	at s.parseMessages (c:\Users\ramar\.vscode\extensions\ms-python.python-2022.12.1\out\client\extension.js:2:525211)
	at s.run (c:\Users\ramar\.vscode\extensions\ms-python.python-2022.12.1\out\client\extension.js:2:506182)
	at processTicksAndRejections (node:internal/process/task_queues:96:5)
	at async s.runLinter (c:\Users\ramar\.vscode\extensions\ms-python.python-2022.12.1\out\client\extension.js:2:524726)]
##########Linting Output - mypy##########

```

When running pylint from a terminal (which I just thought of doing when writing this issue... 😅 ) I get a much more helpful message:

```
$ pylint
usage: pylint [options]
pylint: error: argument --variable-naming-style: invalid choice: 'UPPER_CASE,snake_case' (choose from 'snake_case', 'camelCase', 'PascalCase', 'UPPER_CASE', 'any')
```

I started this issue with the intention of asking whether a single style was allowed, but I have just confirmed that.

With that said, maybe the docs could be rephrased to make this even more obvious? I am used to ESlint for example, and multiple naming conventions by type are  possible, so it's likely that my confusion originated there.

Thanks!

### Documentation for future user

If this is right, I would have expected it to be more explicit in the docs for variable naming styles. Something mentioning that a single style can be applied to each "type", and that in case of requiring multiple styles, then we would need to use regexes.


### Additional context

https://pylint.pycqa.org/en/latest/user_guide/messages/convention/invalid-name.html#cmdoption-variable-naming-style

**Comment by Pierre-Sassoulas** [comment]
Thank you for opening the issue. I'm wondering if it's better to add the feature or document the lack of the feature. Let's wait for other opinions on this. There's also a rework in #7305 to take into account.
